### PR TITLE
SQL React Agent Updates

### PR DESCRIPTION
**Pull request by jmpounders**
Run: `python examples/bank_agent.py`

This new agent is "ReAct" in spirit only.  More efficient and accurate than a forced ReAct-template agent.

- [x] Pre-check for question reasonableness
- [x] Issue #74 
- [x] Issue #75 
- [x] Issue #78 
- [x] Simplify prompt; avoid strict dialogue formatting; don't require function calls

### CODE REVIEW

**Review by kevindewalt — APPROVED** [review]
Looks good based on skimming code, I'll provide additional feedback when I can see it. :-)

**Comment by jmpounders** [comment]
In terms of the agent, this work is something of a placeholder/proof of concept pending a more general agent framework.  Especially with the simplified approach, should be straight forward to pull this into the new framework.

So detailed review of the implementation of the agent in `react.py` may not be warranted since it will be refactored shortly.

File: src/neosophia/agents/react.py [hunk 1]
```diff
@@ -44,33 +51,48 @@ def get_next_message(
 def make_react_agent(
         system_message: str,
         model: Callable,
-        function_descriptions: dict[str, Callable],
-        functions: list[dict],
-        max_llm_calls: int | None = 10
+        function_descriptions: list[dict],
+        functions: dict[str, Callable],
+        max_llm_calls: int | None = 10,
+        simple_formatting: bool | None = True
```
Comment (bdzimmer) [inline src/neosophia/agents/react.py:57]: The way these arguments are used you should remove the `| None`. You don't handle a case where they could take the value `None` so they aren't `Optional` in that sense. (`Optional[T]` means the variable could be `T` or `None`, it doesn't have anything to do with whether the argument has a default value.)

File: src/neosophia/agents/helpers.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    )
+    user_question = f"Here is the transcript of the interaction:\n{transcript}"
+    messages = [
+        openai.Message('system', system_message),
+        openai.Message('user', user_question)
+    ]
+
+    response = model(messages, functions=functions_descriptions)
+
+    return response.content
```
Comment (bdzimmer) [inline src/neosophia/agents/helpers.py:41]: missing newline

**Review by bdzimmer — APPROVED** [review]

File: src/neosophia/agents/react.py (hunk 1 above)
Comment (jmpounders) [reply src/neosophia/agents/react.py:57]: Ah... good call out.

**Pull request merged**